engine/src/main/java/org/hibernate/validator/internal/engine/tracking/PredefinedScopeProcessedBeansTrackingStrategy.java

```diff
@@ -0,0 +1,368 @@
+							directCascadedBeanClasses.add( typeToClassToProcess( bound ) );
+						}
+					}
+					else if ( typeArgument instanceof WildcardType wildcard ) {
+						for ( Type bound : wildcard.getUpperBounds() ) {
+							directCascadedBeanClasses.add( typeToClassToProcess( bound ) );
+						}
+						if ( wildcard.getLowerBounds().length != 0 ) {
+							// if it's a lower bound ? super smth ... it doesn't matter anymore since it can contain anything so go with object ?
+							directCascadedBeanClasses.add( Object.class );
+						}
+					}
+					else {
+						// TODO: instead of failing, add an Object.class and assume it can be anything ?
+						throw new UnsupportedOperationException( typeArgument.getClass().getSimpleName() + " type argument values are not supported." );
+					}
+				}
+			}
+			else {
+				// If we do not have the type arguments then we can go though the value extractors,
+				//  as they are required to define the `@ExtractedValue(type = ???)` ...
+				//  this way we should get the type we want:
+				for ( ValueExtractorDescriptor valueExtractorCandidate : metaData.getValueExtractorCandidates() ) {
+					valueExtractorCandidate.getExtractedType().ifPresent( directCascadedBeanClasses::add );
+				}
+			}
+		}
+
+		for ( ContainerCascadingMetaData sub : metaData.getContainerElementTypesCascadingMetaData() ) {
+			processContainerCascadingMetaData( sub, directCascadedBeanClasses );
+		}
+	}
+
+	private static Class<?> typeToClassToProcess(Type type) {
+		if ( type instanceof Class<?> cascadableClass ) {
+			return cascadableClass;
+		}
+		else if ( type instanceof ParameterizedType parameterizedType ) {
+			return typeToClassToProcess( parameterizedType.getRawType() );
+		}
+		else {
+			// TODO: instead of failing, add an Object.class and assume it can be anything ?
+			//  return Object.class;
+			throw new UnsupportedOperationException( type.getClass().getSimpleName() + " type values are not supported." );
+		}
+	}
+
+	@Override
+	public boolean isEnabledForBean(Class<?> rootBeanClass, boolean hasCascadables) {
+		if ( !hasCascadables ) {
+			return false;
+		}
+
+		return trackingEnabledForBeans.get( rootBeanClass );
+	}
+
+	@Override
+	public boolean isEnabledForReturnValue(Signature signature, boolean hasCascadables) {
+		if ( !hasCascadables ) {
+			return false;
+		}
+
+		return trackingEnabledForReturnValues.getOrDefault( signature, true );
+	}
+
+	@Override
+	public boolean isEnabledForReturnValue(ReturnValueMetaData returnValueMetaData) {
+		if ( !returnValueMetaData.isCascading() ) {
+			return false;
+		}
+
+		Set<Class<?>> directCascadedBeanClasses = new HashSet<>();
+		for ( Cascadable cascadable : returnValueMetaData.getCascadables() ) {
+			processSingleCascadable( cascadable, directCascadedBeanClasses );
+		}
+		for ( Class<?> directCascadedBeanClass : directCascadedBeanClasses ) {
+			if ( trackingEnabledForBeans.get( directCascadedBeanClass ) ) {
+				return true;
+			}
+		}
```

Comment:
the idea here was to reuse the logic from bean tracking detection ^ and go through cascadables, collect what "top level" classes we can cascade into and then use the `trackingEnabledForBeans` map to see if tracking is required ...